# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/nsHttpTransaction.h
# Commit: b3ffe3409712
# Full Hash: b3ffe3409712f31bcf61cf93aa09bf137e45f9bc
# Author: Kershaw Chang <kershaw@mozilla.com>
# Date: 2025-03-29 09:19:43
# Description:
#   Bug 1951554 - Avoid racing on nsHttpTransaction::mConnInfo, r=necko-reviewers,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D243146
# ==============================================================================

diff -r 21cdd98a6828 -r b3ffe3409712 netwerk/protocol/http/nsHttpTransaction.h
--- a/netwerk/protocol/http/nsHttpTransaction.h	Fri Mar 28 17:48:02 2025 +0000
+++ b/netwerk/protocol/http/nsHttpTransaction.h	Fri Mar 28 17:56:38 2025 +0000
@@ -268,6 +268,8 @@
   already_AddRefed<Http2PushedStreamWrapper> TakePushedStreamById(
       uint32_t aStreamId);
 
+  void FinalizeConnInfo();
+
   // IMPORTANT: when adding new values, always add them to the end, otherwise
   // it will mess up telemetry.
   enum HTTPSSVC_CONNECTION_FAILED_REASON : uint32_t {
@@ -578,6 +580,9 @@
   nsCOMPtr<nsITimer> mFastFallbackTimer;
   nsCOMPtr<nsITimer> mHttp3BackupTimer;
   RefPtr<nsHttpConnectionInfo> mBackupConnInfo;
+  // A clone of mConnInfo taken when this transaction is activated.
+  // Describes the server that the associated connection is connected to.
+  RefPtr<nsHttpConnectionInfo> mFinalizedConnInfo;
   RefPtr<HTTPSRecordResolver> mResolver;
   TRANSACTION_RESTART_REASON mRestartReason = TRANSACTION_RESTART_NONE;
 
